src/nvim/ex_cmds.c

```diff
@@ -5846,3 +5914,286 @@ void set_context_in_sign_cmd(expand_T *xp, char_u *arg)
     }
   }
 }
+
+/// function called after a live command to get back to
+/// a normal state
+/// validate can take two values :
+///     0 if the live_command has not been validated
+///     1 if the user has validated the command
+void finish_live_cmd(int save_state,
+                     garray_T *winsizes,
+                     int save_exmode,
+                     int save_restart_edit,
+                     int save_cmdmsg_rl,
+                     int validate) {
+  block_autocmds();
+
+  if (validate == 0 && sub_done == 1) {
+    do_cmdline_cmd(":u");
+    sub_done = 0;
+  }
+
+  // Restore window sizes.
+  if (winsizes != NULL) {
+    win_size_restore(winsizes);
+    ga_clear(winsizes);
+  }
+
+  unblock_autocmds();
+
+  char_u typestr[2];
+  typestr[0] = cmdwin_type;
+  typestr[1] = NUL;
+  apply_autocmds(EVENT_CMDWINLEAVE, typestr, typestr, false, curbuf);
+
+  exmode_active = save_exmode;
+  restart_edit = save_restart_edit;
+  cmdmsg_rl = save_cmdmsg_rl;
+
+  cmdwin_type = 0;
+  RedrawingDisabled = 0;
+
+  State = save_state;
+  setmouse();
+  update_screen(0);
+}
+
+/// ex_window_inc_sub()
+/// Open a window for future displaying of the inc_sub mode.
+///
+/// Does not allow editing in the window.
+/// Returns when the window is closed.
+///
+/// @param pat the pattern word
+/// @param sub the replacement word
+/// @param lmatch the list containing our data
+void ex_window_inc_sub(char_u * pat,
+                       char_u * sub,
+                       klist_t(matchedline_T) *lmatch, // NOLINT
+                       bool split)
+FUNC_ATTR_NONNULL_ARG(1, 2, 3)
```

Comment:
This should be indented. Function name in doc comment should not be present. Also no NOLINT.

By the way, another way to avoid linter is

    typedef klist_t(matchedline_T) MatchedLinesList;

: if line does not start with a space this rule does not apply (workaround for `int func(void) FUNC_ATTR_WARN_UNUSED_RESULT`).